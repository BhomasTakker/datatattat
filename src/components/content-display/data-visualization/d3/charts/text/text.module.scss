@import "../../styles/colors.scss";

.text {
}

.label {
	fill: $textMain;
	font-size: 1.5em;
	text-transform: capitalize;
}

.tick {
	fill: $textLight;
	font-size: 1em;
	cursor: default;
}

.title {
}

.legend {
	fill: $textLight;
	font-size: 2.5em;
	text-transform: capitalize;
}
