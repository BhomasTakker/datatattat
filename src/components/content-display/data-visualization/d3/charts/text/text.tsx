import { SVGProps } from "react";
import styles from "./text.module.scss";

type TextVariant = "label" | "tick" | "title";

export type Text = {
	variant: TextVariant;
	text: string;
} & SVGProps<SVGTextElement>;

export const Text = ({ text, variant, ...rest }: Text) => {
	return (
		<text {...rest} className={styles[variant]}>
			{text}
		</text>
	);
};
