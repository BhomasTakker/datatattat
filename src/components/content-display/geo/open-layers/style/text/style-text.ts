import Text from "ol/style/Text";

export type StyleText = "Emoji";

type CreateStyleEmoji = {
	text: string;
	scale?: number;
	rotation?: number;
};
// Maybe in a text thing
// getLabel etc
export const createStyleEmoji = ({
	text,
	scale = 1,
	rotation,
}: CreateStyleEmoji) =>
	new Text({
		text,
		justify: "center",
		scale,
		rotation,
	});
type AvailableText = CreateStyleEmoji;
export type CreateText = {
	type: StyleText;
} & AvailableText;

export const createStyleText = ({ type, ...rest }: CreateText) => {
	switch (type) {
		case "Emoji":
			return createStyleEmoji(rest as CreateStyleEmoji);
	}
};
