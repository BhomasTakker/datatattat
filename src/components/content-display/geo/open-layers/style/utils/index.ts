import Feature from "ol/Feature";
import { Geometry } from "../../types/open-layers.types";

export const filterFeature = (feature: Feature, geometry: Geometry) => {
	const featureGeometry = feature.getGeometry()?.getType();
	console.log({ featureGeometry });
	return featureGeometry == geometry;
};
