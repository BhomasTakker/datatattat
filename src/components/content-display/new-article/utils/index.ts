type ClassName = string | undefined | null;

export const addCssClass = (className: ClassName) => {
	return className ? ` ${className}` : "";
};

export const addCssClasses = (classNames: ClassName[]) => {
	return classNames.filter((name) => name).join(" ");
};
