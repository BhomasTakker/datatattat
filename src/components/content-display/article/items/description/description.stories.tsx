import type { Meta, StoryObj } from "@storybook/react";
import { Description } from "./description";
import { LOREM_5_S } from "mockData/text/lorem";

const meta: Meta<typeof Description> = {
	component: Description,
};
export default meta;
type Story = StoryObj<typeof Description>;

export const Standard: Story = {
	args: {
		description: LOREM_5_S,
		maxLines: 3,
		maxWidth: "100%",
		minWidth: "400px",
	},
};
