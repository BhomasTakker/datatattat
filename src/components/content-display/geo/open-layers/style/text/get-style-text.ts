import { EmojiMap } from "../../filters/types";
import { applyEmojiMap } from "../apply-emoji-map";
import { CreateText } from "./style-text";

type GetStyleText = {
	feature: { [x: string]: unknown };
	emojiMap?: EmojiMap;
};

export const getStyleText = ({
	feature,
	emojiMap,
}: GetStyleText): CreateText | null => {
	switch (true) {
		// check if given text
		case !!emojiMap:
			return applyEmojiMap({ shapeMap: emojiMap, object: feature });
		default:
			return null;
	}
};
