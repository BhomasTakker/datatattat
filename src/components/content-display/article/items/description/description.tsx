import { Typography } from "@mui/material";
import { CSS } from "@/src/css/text";
import { stripHTML } from "@/src/utils/html";
import { log } from "@/src/lib/logger";
import { TypographyProps } from "@/src/types/mui";

export type DescriptionVariants = "Primary" | "Secondary";

export type DescriptionProps = {
	description: string;
	maxLines?: number;
	// maxWidth?: string; // how to do number px | % etc
	// minWidth?: string;
	// variant: DescriptionVariants;
} & TypographyProps;

export const Description = ({
	description,
	maxLines = 3,
	// variant = "Primary",
	// look at this - we probably don't want defaults?
	// maxWidth = "100%",
	// 400px AND 100% to be taken from a widths
	// i.e. SIZES.FULL, SIZES.PRESET_400
	// minWidth = "400px",
	...rest
}: DescriptionProps) => {
	log({ code: "FEATURE:0010", context: "DESCRIPTION" });
	return (
		<Typography
			data-testid="description"
			sx={{
				// width object / max, min, breakpoint sizes?
				// ...minMaxWidth({ maxWidth, minWidth }), // do a width object
				...CSS.maxLines({ maxLines }),
			}}
			{...rest}
			// This would be a no? you could go with text variant
			// font family, size, color, etc
			{...CSS.textStyle()}
		>
			{stripHTML(description)}
		</Typography>
	);
};
