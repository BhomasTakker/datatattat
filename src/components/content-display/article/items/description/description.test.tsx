import { render, screen } from "@testing-library/react";
import { Description } from "./description";
import { LOREM_3_S } from "mockData/text/lorem";
import { stripHTML } from "@/src/utils/html";
import { CSS } from "@/src/css/text";

// const stripHTMLSpy = jest.spyOn(require("@/src/utils/html/index"), "strpHTML");
jest.mock("@/src/utils/html", () => {
	return {
		...jest.requireActual("@/src/utils/html"),
		stripHTML: jest.fn(),
	};
});

(stripHTML as jest.Mock).mockImplementation((str: string) => str);

const maxLinesSpy = jest.spyOn(CSS, "maxLines");

describe("Description", () => {
	beforeEach(() => {
		(stripHTML as jest.Mock).mockClear();
		maxLinesSpy.mockClear();
	});

	it("should render Description", () => {
		render(<Description description={LOREM_3_S} />);
		const descriptionElement = screen.getByTestId("description");
		expect(descriptionElement).toBeInTheDocument();
	});

	it("should render expected text", () => {
		render(<Description description={LOREM_3_S} />);
		const descriptionElement = screen.getByText(LOREM_3_S);
		expect(descriptionElement).toBeInTheDocument();
	});

	describe("stripHTML", () => {
		it("should call stripHTML once", () => {
			render(<Description description={LOREM_3_S} />);
			expect(stripHTML).toHaveBeenCalledTimes(1);
		});
	});

	describe("snaps", () => {
		// Do a snap for default
		// and each prop or variant etc
		it("should match snapshot", () => {
			render(<Description description={LOREM_3_S} />);
			const descriptionElement = screen.getByTestId("description");
			expect(descriptionElement).toMatchSnapshot();
		});

		it("should render expected number of lines", () => {
			render(<Description description={LOREM_3_S} maxLines={3} />);
			const descriptionElement = screen.getByTestId("description");
			expect(descriptionElement).toMatchSnapshot();
		});
	});

	describe("max lines", () => {
		it("should render expected number of lines", () => {
			render(<Description description={LOREM_3_S} maxLines={3} />);
			const descriptionElement = screen.getByTestId("description");
			expect(maxLinesSpy).toHaveBeenCalledTimes(1);
			expect(maxLinesSpy).toHaveBeenCalledWith({ maxLines: 3 });
		});
	});
});
