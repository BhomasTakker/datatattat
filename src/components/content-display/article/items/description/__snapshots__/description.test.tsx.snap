// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Description snaps should match snapshot 1`] = `
<span
  class="MuiTypography-root MuiTypography-body1 css-2u148c-MuiTypography-root"
  data-testid="description"
>
  Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Tellus orci ac auctor augue mauris. Sit amet nulla facilisi morbi.
</span>
`;

exports[`Description snaps should render expected number of lines 1`] = `
<span
  class="MuiTypography-root MuiTypography-body1 css-2u148c-MuiTypography-root"
  data-testid="description"
>
  Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Tellus orci ac auctor augue mauris. Sit amet nulla facilisi morbi.
</span>
`;
